cni/network/network.go

```diff
@@ -1143,3 +1193,47 @@ func determineSnat() (bool, bool, error) {
 
 	return snatConfig.EnableSnatForDns, snatConfig.EnableSnatOnHost, nil
 }
+
+func convertNnsToCniResult(
+	netRes *nnscontracts.ConfigureContainerNetworkingResponse,
+	ifName string,
+	podName string,
+	operationName string) *cniTypesCurr.Result {
+
+	// This function does not add interfaces to CNI result. Reason being CRI (containerD in baremetal case)
+	// only looks for default interface named "eth0" and this default interface is added in the defer
+	// method of ADD method
+	result := &cniTypesCurr.Result{}
+	var resultIpconfigs []*cniTypesCurr.IPConfig
+
+	if netRes.Interfaces != nil {
+		for i, ni := range netRes.Interfaces {
+
+			intIndex := i
```

Comment:
why do we need copy of this variable?

Reply:
I think it being done when creating result object. That's why.

Reply:
i think this is the place you were mentioning:` Interface: &intIndex,` . can't we use &i instead of &intIndex?